### PR TITLE
Check sp_list_port return value

### PR DESCRIPTION
**Pull request by Coimbra1984**
Prevent null pointer exception on android in case of permsiision denied on serial ports

### CODE REVIEW

File: lib/src/port.dart [hunk 1]
```diff
@@ -226,8 +226,10 @@ class _SerialPortImpl implements SerialPort {
   }
 
   static List<String> get availablePorts {
+    int rv;
     final out = ffi.allocate<ffi.Pointer<ffi.Pointer<sp_port>>>();
-    Util.call(() => dylib.sp_list_ports(out));
+    Util.call(() => rv = dylib.sp_list_ports(out));
```
Comment (jpnurmi) [inline lib/src/port.dart:231]: In #8 it was mentioned that `sp_list_ports()` returned `SP_ERR_FAIL` (-2). Wouldn't `Util.call()` throw a `SerialPortError` for that?

File: lib/src/port.dart [hunk 2]
```diff
@@ -226,8 +226,10 @@ class _SerialPortImpl implements SerialPort {
...
+    if (rv != sp_return.SP_OK) return [];
```
Comment (jpnurmi) [inline lib/src/port.dart:232]: The `ffi.free()` calls below need to be executed to avoid leaking resources allocated above.

**Review by jpnurmi — COMMENTED** [review]
Hi, thanks for helping out!

File: lib/src/port.dart (hunk 1 above)
Comment (Coimbra1984) [reply lib/src/port.dart:231]: No, for me on my android phone it did not throw an error

File: lib/src/port.dart (hunk 2 above)
Comment (Coimbra1984) [reply lib/src/port.dart:232]: you are right!

**Review by jpnurmi — APPROVED** [review]

**Comment by jpnurmi** [comment]
v0.0.5 with the fix published: https://pub.dev/packages/dart_serial_port/versions/0.0.5

**Pull request merged**